### PR TITLE
Updated Strings

### PR DESCRIPTION
**Pull request by svenkat19**
Do let me know if there are any issues. Cheers!!

### CODE REVIEW

**Comment by ianpaul** [comment]
@svenkat19 did you run Flashbake and test that all of this works?

**Comment by svenkat19** [comment]
> @svenkat19 did you run Flashbake and test that all of this works?

I am kinda lost on how to run FLashbake. I am quite new to this. Could you help me out?

**Comment by ianpaul** [comment]
> > @svenkat19 did you run Flashbake and test that all of this works?
> 
> I am kinda lost on how to run FLashbake. I am quite new to this. Could you help me out?

Sure. This is a command line program for Unix-like systems so you need to be working out of a terminal on Linux, Mac, WSL on Windows 10, or Crostini on Chrome OS. We have instructions for all operating systems on the [Wiki](https://github.com/cmdln/flashbake/wiki/Installation). 

If you need more help after that let me know. If you're using Mac note that you also need to install each dependency listed in the `tox.ini` file manually. 

Once you've got it installed, you need to set-up a sample directory with git initialized, a configuration file (also explained in the wiki) with every single plugin that you changed the code for, and a test.txt hotfile. Then run flashbake with the two test flags: `flashbake -c ~/your_sample_directory`, and then `flashbake -d ~/your_sample_directory`. Then to be extra sure you can run a live commit with `flashbake ~/your_sample_directory`, and then look at the commit history with `git log` to see that everything's ok. 

If you can't get Flashbake running let me know (it took me a few tries my first time) and I'll help further. 

Once you've checked that everything is running properly, I will too, and then we'll move forward from there.

**Comment by svenkat19** [comment]
> > > @svenkat19 did you run Flashbake and test that all of this works?
> > 
> > 
> > I am kinda lost on how to run FLashbake. I am quite new to this. Could you help me out?
> 
> Sure. This is a command line program that helps technically adept writers automatically commit their work to a local git repository every _n_ minutes. Flashbake is for Unix-like systems so you need to be working out of a terminal on Linux, Mac, WSL on Windows 10, or Crostini on Chrome OS. We have installation instructions for all operating systems on the [Wiki](https://github.com/cmdln/flashbake/wiki/Installation).
> 
> If you need more help after that let me know. If you're using Mac note that you also need to install the two dependencies listed in the `tox.ini` file manually.
> 
> Once you've got it installed, you need to set-up a sample directory with git initialized, a configuration file (also explained in the wiki) with every plugin that you changed the code for, and a `test.txt` hot file (we call the files that Flashbake puts into a git commit "hot files"). Then run flashbake with the two test flags: `flashbake -c ~/your_sample_directory`, and then `flashbake -d ~/your_sample_directory`. Then to be extra sure you can run a live commit with `flashbake ~/your_sample_directory`, and then look at the commit history with `git log` to see that everything's ok.
> 
> If you can't get Flashbake running let me know (it took me a few tries my first time) and I'll help further.
> 
> Once you've checked that everything is running properly, I will test it as a second check that everything's ok, and then we'll move forward from there.

image
I installed WSL and was able to run the setup. I am stuck at the configuration file . Could you please elaborate more on that? Thanks.

**Comment by ianpaul** [comment]
Sure. Here's a sample config file below. You can just copy and paste it to get started. If you can't test the plugins that require getting API keys like weather, lastfm, and Twitter, don't worry about it. I'll handle testing those.

Remember that if you run a full test without a `-c` or `-d` flag you'll need to add some extra text to your `story.txt` file so that Git will notice a change and Flashbake will have something to commit. 

**Sample config file:**

#My hot files
story.txt

#My plugins
plugins:flashbake.plugins.timezone:TimeZone

plugins:flashbake.plugins.location:Location

plugins:flashbake.plugins.timestamp:Timestamp
timestamp_time_format:local
timestamp_time_hours: 24

plugins:flashbake.plugins.feed:Feed
feed_url: https://www.pcworld.com/author/Ian-Paul/
# feed_author: Ian Paul 
feed_limit: 3

**Comment by svenkat19** [comment]
> Sure. Here's a sample config file below. You can just copy and paste it to get started. If you can't test the plugins that require getting API keys like weather, lastfm, and Twitter, don't worry about it. I'll handle testing those.
> 
> Remember that if you run a full test without a `-c` or `-d` flag you'll need to add some extra text to your `story.txt` file so that Git will notice a change and Flashbake will have something to commit.
> 
> **Sample config file:**
> 
> #My hot files
> story.txt
> 
> #My plugins
> plugins:flashbake.plugins.timezone:TimeZone
> 
> plugins:flashbake.plugins.location:Location
> 
> plugins:flashbake.plugins.timestamp:Timestamp
> timestamp_time_format:local
> timestamp_time_hours: 24
> 
> plugins:flashbake.plugins.feed:Feed
> feed_url: https://www.pcworld.com/author/Ian-Paul/
> feed_author: Ian Paul
> feed_limit: 3

So I created a sample directory , initialized git.  Can you review this?
image

How do I run Flashbake?

**Comment by ianpaul** [comment]
So try three things:

1. run  `flashbake -d -v /mnt/d/planets/test/` (I am just guessing at the file path) This will do a dry run and you'll see some of the behind the scenes log messages. Check for error messages and that everything runs smoothly.
2. then run `flashbake -c /mnt/d/planets/test/`  This will tell Flashbake to only create the git commit message as a test and print the commit message to the terminal. You want to check to make sure that the commit message contains the data you expect and it doesn't raise any errors.
3. Finally, run `git status` in the flashbake test directory to make sure there are file(s) waiting to be committed. Then run a live update with `flashbake /mnt/d/planets/test/` Then check to see that the file(s) were committed as expected. 

If everything works as expected let me know here, and I'll take a look.

**Comment by svenkat19** [comment]
> 1. flashbake -d -v /mnt/d/planets/test/

image

I am getting this error

**Comment by ianpaul** [comment]
Okay, so this is where error analysis comes in. If you look at the LAST error message (that's usually the key). You'll see that the error is on line 121 of __init__.py. And if you look at your commit history, you changed a few of the lines before it. So what you can do (assuming this is the only problem) is that you can change it back to what it was and see if you get the error again. If you don't you know there's a problem with your code. If you do get the same error then it's something else. But I'm willing to bet it's something to do with your changes. Always, always look at the last line of an error message in Python to start figuring out what went wrong. Also, Stackoverflow is your friend in situations like this. Though syntax errors are sometimes harder to decode on the forums.

**Comment by svenkat19** [comment]
I had only one syntax error and I fixed it.
I redownloaded the master and ran it, it gave the same error. So I suppose there is an issue with master itself. 
image
Can you run the master and see if it's an issue with the program or my system is messed up

**Comment by ianpaul** [comment]
I just did a fresh install from the source, and ran my test folder with no issues. I'd suggest this. And I'm sorry, this is my fault I should've been more explicit about how to work. Here's what I'd like you to do:

1. Copy all the changes you've made into a separate folder. Somewhere where you won't lose the changes so you don't have to redo too much of the work.
2. Erase Flashbake from your PC, go to Github and delete your copy of the repo, and then re-fork it. 
3. Next do a fresh git clone from the newly forked repo, and reinstall Flashbake on your PC.
4. Now run Flashbake to make sure everything works in its original state. 
5. Now create a new branch for your changes. 
6. In that new branch make one change at a time (or at the most one file at a time). Then reinstall Flashbake from your branch so it has the small number of changes, and run Flashbake. If your change doesn't raise any errors, commit the change to your branch, and go to step 8. If you had errors go to step 7. 
7. If your change raises an error, then read the last line of the error message and try to fix it by searching for the error via Google. This will inevitably lead to TONS of stackoverflow answers. Find one that seems to apply to your situation and try to adapt it to fix your issue (this is an incredible way to learn. It's how I learn). Remember that Python fails in a cascade (one failure leads to another, leads to another) so almost always you only have to read the last line or last two or three lines to find the error. Once you've fixed the error, commit to your branch and move to step 8.
8. If your change doesn't raise an error, then follow the process from step 6 again, and make another change (or at the most change one file at a time). Then reinstall Flashbake again with your new changes, run Flashbake, if the next set of changes work, then commit, and then keep going. 

This process is the way to work. It will ultimately save you time, and it's how I do things even with something as seemingly simple as print statements. 

To sum up: Do small changes, test them, and once they work commit the changes to your branch, and then repeat that process until your task is complete.

**Comment by svenkat19** [comment]
> I just did a fresh install from the source, and ran my test folder with no issues. I'd suggest this. And I'm sorry, this is my fault I should've been more explicit about how to work. Here's what I'd like you to do:
> 
> 1. Copy all the changes you've made into a separate folder. Somewhere where you won't lose the changes so you don't have to redo too much of the work.
> 2. Erase Flashbake from your PC, go to Github and delete your copy of the repo, and then re-fork it.
> 3. Next do a fresh git clone from the newly forked repo, and reinstall Flashbake on your PC.
> 4. Now run Flashbake to make sure everything works in its original state.
> 5. Now create a new branch for your changes.
> 6. In that new branch make one change at a time (or at the most one file at a time). Then reinstall Flashbake from your branch so it has the small number of changes, and run Flashbake. If your change doesn't raise any errors, commit the change to your branch, and go to step 8. If you had errors go to step 7.
> 7. If your change raises an error, then read the last line of the error message and try to fix it by searching for the error via Google. This will inevitably lead to TONS of stackoverflow answers. Find one that seems to apply to your situation and try to adapt it to fix your issue (this is an incredible way to learn. It's how I learn). Remember that Python fails in a cascade (one failure leads to another, leads to another) so almost always you only have to read the last line or last two or three lines to find the error. Once you've fixed the error, commit to your branch and move to step 8.
> 8. If your change doesn't raise an error, then follow the process from step 6 again, and make another change (or at the most change one file at a time). Then reinstall Flashbake again with your new changes, run Flashbake, if the next set of changes work, then commit, and then keep going.
> 
> This process is the way to work. It will ultimately save you time, and it's how I do things even with something as seemingly simple as print statements.
> 
> To sum up: Do small changes, test them, and once they work commit the changes to your branch, and then repeat that process until your task is complete.

I did a fresh install of FlashBake and it is giving me the same error :( ! I checked my WSL again, and I had no issues running other tasks on the terminal.

**Comment by ianpaul** [comment]
hmmmm, that shouldn't be happening. I will check on a Windows PC and see what I get. I'll get back to you soon.

**Comment by ianpaul** [comment]
So I just did a fresh install of WSL on a laptop I have lying around running Windows 10, and everything went fine. Let's go through a few steps. First, which distribution are you running in WSL, Ubuntu? Also, can you run `python3 --version` and tell me which version of Python you're running?

**Comment by svenkat19** [comment]
> So I just did a fresh install of WSL on a laptop I have lying around running Windows 10, and everything went fine. Let's go through a few steps. First, which distribution are you running in WSL, Ubuntu? Also, can you run `python3 --version` and tell me which version of Python you're running?

UBUNTU 20.04 LTS and Python 3.9.6

**Comment by ianpaul** [comment]
Okay, that should all be fine. Let's get two more pieces of information. Can you run the following code in WSL and tell me the output? `pip3 list | grep "enum"`

Also, can you copy and paste your Flashbake configuration file in here?

**Comment by ianpaul** [comment]
Also, please run `ls -a` in your test folder and tell me the names of all the files in your folder.

**Comment by svenkat19** [comment]
image

image

image

**Comment by ianpaul** [comment]
Okay please go into your test folder and run this `mv story.txt.txt story.txt` This will rename your story file to the correct name in the config file. Then run Flashbake again and see if you get the errors.

**Comment by svenkat19** [comment]
I am getting the same error even after I ran `mv story.txt.txt story.txt`
image

**Comment by ianpaul** [comment]
huh, okay, let's try this. Change your config to this. This tells Flashbake to only run with the Location plugin. Save it and run Flashbake again.

#My hot files
story.txt

#My plugins
#plugins:flashbake.plugins.timezone:TimeZone

plugins:flashbake.plugins.location:Location

#plugins:flashbake.plugins.timestamp:Timestamp
#timestamp_time_format: local
#timestamp_time_hours: 24

#plugins:flashbake.plugins.feed:Feed
#feed_url: https://www.pcworld.com/author/Ian-Paul/
#feed_author: Ian Paul
#feed_limit: 3

**Comment by svenkat19** [comment]
> huh, okay, let's try this. Change your config to this. This tells Flashbake to only run with the Location plugin. Save it and run Flashbake again.
> 
> #My hot files
> story.txt
> 
> #My plugins
> #plugins:flashbake.plugins.timezone:TimeZone
> 
> plugins:flashbake.plugins.location:Location
> 
> #plugins:flashbake.plugins.timestamp:Timestamp
> #timestamp_time_format: local
> #timestamp_time_hours: 24
> 
> #plugins:flashbake.plugins.feed:Feed
> #feed_url: https://www.pcworld.com/author/Ian-Paul/
> #feed_author: Ian Paul
> #feed_limit: 3

I changed it. I am getting the same error. Should I try running it on a different PC?

**Comment by ianpaul** [comment]
If you have one, yes let's try that and see what happens.

**Comment by svenkat19** [comment]
> If you have another one, yes let's try that and see what happens.

I tried on two  VMs, tested it on Windows 10 and Windows 11(dev), I am getting the same issue 😬

**Comment by ianpaul** [comment]
Huh, I assume you're installing from the same repo each time. Let's try this. Do `sudo rm -r flashbake` in a VM or one of your PCs it doesn't matter to get rid of Flashbake from that PC entirely. Then download the ZIP file here: https://github.com/ianpaul/flashbake/releases/tag/v0.30-beta.1. Unzip it, and before you install make sure you have Python3 and python3-setuptools installed (be sure to install setuptools from apt with `sudo apt install python3-setuptools`, not pip3). Then run `sudo python3 setup.py install` as usual.

**Comment by ianpaul** [comment]
So I just tried flashbake again in a VM with Windows 10 and I can't recreate your error. Even with python 3.9 it works out. I did run into a permissions issue, but that's solved with `sudo` and I'd never seen that before on Windows anyway. Just to review are you installing like this on a fresh machine after enabling WSL?
1. `sudo apt-get update`
2. `sudo apt-get upgrade -y`
3. `git clone https://github.com/svenkat19/flashbake.git`
4. `cd flashbake`
5. `sudo python3 setup.py install`

**Comment by svenkat19** [comment]
I did reinstall Ubuntu, but haven't done so with Python. What I'll do , though logically it shouldn't make any difference, is I'll wipe out Python , WSL and Flashbake, redownload everything, and then try again. Plus I'll try it on another PC as well. I'll get back to you in an hour or two. I'll also try using Ubuntu 18 and Arch. Sounds good?

**Comment by svenkat19** [comment]
> I did reinstall Ubuntu, but haven't done so with Python. What I'll do , though logically it shouldn't make any difference, is I'll wipe out Python , WSL and Flashbake, redownload everything, and then try again. Plus I'll try it on another PC as well. I'll get back to you in an hour or two. I'll also try using Ubuntu 18 and Arch. Sounds good?

**Comment by ianpaul** [comment]
That's too complicated, I think. Wiping out Python shouldn't make any difference at all. Plus Python comes built in to Ubuntu and wiping it out might cause malfunctions. Look, something unusual that is causing this and we're trying to find out what it is. How about you try from a fresh WSL Ubuntu 20.04 install. Then follow the exact 5 instructions I've laid out for you above, and see what that gets you.

**Comment by ianpaul** [comment]
Also, if following those instructions causes the same error. Then try running Flashbake with no plugins at all. Just commits to the hot file.

**Comment by svenkat19** [comment]
image
Looks like it worked

**Comment by ianpaul** [comment]
Awesome! Okay, now that we know it works. Try a real flashbake run without the `-c` and then check `git log` to see that the commit happened. Then if that works. Start a branch and slowly add your changes in small bits to make sure they work!

**Comment by svenkat19** [comment]
image
Is this okay?

**Comment by svenkat19** [comment]
I kinda cheated a little. I ran my changes directly and it ran without a hitch. 
image

**Comment by ianpaul** [comment]
Okay, great! Usually we'd like you to do small changes, but since this pull request is already done I'll test it out and get back to you. I probably won't get to it before the weekend. Thanks again!

**Comment by svenkat19** [comment]
Well I directly ran it because it was just updating strings. Any issues do let me know :). Cheers

**Comment by ianpaul** [comment]
Yeah, you'd be surprised how many times a print statement change can cause problems.

**Comment by ianpaul** [comment]
Hi @svenkat19 after all this forking and re-forking I don't see the changes in this pull request in your GitHub fork anymore. I am going to close this pull request. Can you please commit your changes, push them to Github, and open another pull request?  Before you do though, please fetch upstream as your fork is one commit behind the main repo. Thanks!
